app/services/no_bp_measure_service.rb

```diff
@@ -0,0 +1,88 @@
+class NoBPMeasureService
+  CACHE_VERSION = 1
+
+  def initialize(region, periods:, type: :missed_visits)
+    @region = region
+    @periods = periods
+    @facilities = region.facilities.to_a
+    @type = type
+  end
+
+  attr_reader :facilities
+  attr_reader :periods
+  attr_reader :region
+  attr_reader :type
+
+  def call
+    Rails.cache.fetch(cache_key, version: cache_version, expires_in: 7.days, force: force_cache?) do
+      periods.each_with_object({}) do |period, result|
+        result[period] = missed_visits_for(period)
+      end
+    end
+  end
+
+  def missed_visits_for(period)
+    if type == :missed_visits
+      visit_start_range = period.advance(years: -1).start_date
+      visit_end_range = period.advance(months: -3).start_date
+    else
+      visit_start_range = "-infinity"
+      visit_end_range = period.advance(years: -1).start_date
+    end
+    bind_attributes = {
+      hypertension: "yes",
+      facilities: facilities.map(&:id),
+      visit_start_range: visit_start_range,
+      visit_end_range: visit_end_range,
+      bp_start_range: period.advance(months: -3).start_date,
+      bp_end_range: period.end_date
```

Comment:
Once we remove the ability to see the current month, this can be changed to just be start_date to start_date.  As it stands, this could be _almost_ four months of control data.